boto/s3/keyfile.py

```diff
@@ -84,9 +84,17 @@ def seek(self, pos, whence=os.SEEK_SET):
 
     self.location = pos
 
-  def read(self, size):
-    self.location += size
-    return self.key.read(size)
+  def read(self,*args):
+    if len(args) == 0 or (len(args) == 1 and args[0] < 0):
+      rv = self.key.read()
+      if rv != None:
+	self.location += len(rv)
+    else:
```

Comment:
Not sure this branching is required. If `key` follows `file` protocol we can pass negative values too. 
So the resulted should be like

```
result = self.key.read(size)
if result is not None:
    self.location  += len(result)
return result
```